spikingjelly/activation_based/op_counter/analytical_energy/core.py

```
@@ -0,0 +1,484 @@
+            self._lemaire_tracker.attach(self.model)
+            self._addr_estimator.attach(self.model)
+        return self
+
+    def __exit__(self, exc_type, exc, tb):
+        self._active = False
+        try:
+            self._lemaire_tracker.remove()
+            self._addr_estimator.remove()
+        finally:
+            self._current_stage_name = None
+            return self._dispatch_mode.__exit__(exc_type, exc, tb)
```

Ruff (0.15.12)

[warning] `return` inside `finally` blocks cause exceptions to be silenced

(B012)